### ISSUE TITLE
`http.status_code` tag in the transaction payload

### ISSUE THREAD

**Issue by marandaneto** [getsentry/develop#445]
![unknown](https://user-images.githubusercontent.com/5731772/135980479-98ae68b4-1abd-44f6-acf4-7c0d16f8fd09.png)

The UI special case the `http.status_code` tag in the transaction payload (not spans) to show the status sublabel with the HTTP status code (see flask image).

This looks suboptimal due to, let's say we have a transaction that finishes with the status OK, this transaction has 2 spans which are HTTP requests, one of them is a 200 and the other one is a 500, both are running in parallel and since the tag is in the transaction tags and not span tags, the span that finishes the later always wins setting the tag, overwriting the previous tag.

The same transaction may have a 200 or 500 randomly and can be confusing.

is it an issue? should we address that? also, `http.status_code` isn't part of the spec so most SDKs are not setting it.

**Comment by rhcarvalho** [comment]
For web servers, on the transaction level, the HTTP status is the status of the incoming request -- and not of any outgoing requests (child spans).

Does that clarify the case you are describing?

**Comment by marandaneto** [comment]
@rhcarvalho mmmm, that makes sense, since each incoming request will be its own transaction.

should we doc this special tag @rhcarvalho ?

**Comment by rhcarvalho** [comment]
> should we doc this special tag @rhcarvalho ?

Yes, let's document it, it is used in the Sentry frontend and more SDKs could report it.

- https://github.com/getsentry/sentry/blob/9707fd3f98fa51d5d363cbe34b59aa51588db246/static/app/components/events/rootSpanStatus.tsx#L48
- https://github.com/getsentry/sentry/blob/0e372653c1ab02139718f29e0903e12e452f1f88/static/app/views/performance/transactionDetails/eventMetas.tsx#L230

And apparently that [expectation holds for more than two years](https://github.com/getsentry/sentry/blame/be0c24a56353b04ade8011923da397607a5ed45c/src/sentry/interfaces/spans.py#L30), even before GA.

**Comment by marandaneto** [comment]
we only document the HTTP client integrations https://develop.sentry.dev/sdk/features/#http-client-integrations but not Web servers, or is it documented somewhere else?

it should not be that different from the client one tho.

**Comment by adinauer** [comment]
Should be resolved by https://github.com/getsentry/sentry-java/pull/2287


https://github.com/getsentry/sentry-java/blob/ba4957762dc4c113ed21c09398a94deec7588407/sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt#L196-L205

**Comment by marandaneto** [comment]
Nope, this is not the same thing, this issue is about transactions and not events.

**Comment by marandaneto** [comment]
Cocoa also sends this tag for spans at least, I believe it might be related to https://github.com/getsentry/sentry-cocoa/blob/1453a8aaf8141ad4c304e723f1f17a674e2a839f/Sources/Sentry/SentryNetworkTracker.m#L250-L251

**Comment by adinauer** [comment]
For Spring WebMVC we're currently unable to get the correct response code. For Spring WebFlux we've added the response code to `contexts/response/status_code` for transactions.

We have to check if the UI has already been updated to  also look at `http.response.status_code` in Span data and `contexts/response/status_code` for transactions.

**Comment by adinauer** [comment]
Just confirmed, UI does not look at `contexts/response/status_code`.

**Comment by adinauer** [comment]
Also not `http.response.status_code` in Span data

**Comment by adinauer** [comment]
We do not want to look at span data here as the status is supposed to display the status of the transaction.

Opened https://github.com/getsentry/sentry/issues/54274 for reading the status code from response context in the UI.

Once the UI can show status code from there we can do start changing SDKs.

- [ ] We also have to document where to put the status code for transactions.
